controllers/ibmpowervscluster_controller.go

```diff
@@ -111,142 +113,228 @@ func (r *IBMPowerVSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Re
+	for val := range ch {
+		if val.Requeue {
+			requeue = true
+		}
+		if val.error != nil {
+			errList = append(errList, val.error)
+		}
+	}
+
+	if requeue && len(errList) > 1 {
+		return ctrl.Result{RequeueAfter: 30 * time.Second}, kerrors.NewAggregate(errList)
+	} else if requeue {
+		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
+	} else if len(errList) > 1 {
+		return ctrl.Result{}, kerrors.NewAggregate(errList)
+	}
 
 	// reconcile Transit Gateway
 	clusterScope.Info("Reconciling Transit Gateway")
 	if requeue, err := clusterScope.ReconcileTransitGateway(); err != nil {
 		clusterScope.Error(err, "failed to reconcile transit gateway")
-		conditions.MarkFalse(powerVSCluster, infrav1beta2.TransitGatewayReadyCondition, infrav1beta2.TransitGatewayReconciliationFailedReason, capiv1beta1.ConditionSeverityError, err.Error())
+		conditions.MarkFalse(updatePowerVSCluster.cluster, infrav1beta2.TransitGatewayReadyCondition, infrav1beta2.TransitGatewayReconciliationFailedReason, capiv1beta1.ConditionSeverityError, err.Error())
 		return reconcile.Result{}, err
 	} else if requeue {
 		clusterScope.Info("Transit gateway creation is pending, requeuing")
 		return reconcile.Result{RequeueAfter: 1 * time.Minute}, nil
 	}
-	conditions.MarkTrue(powerVSCluster, infrav1beta2.TransitGatewayReadyCondition)
-
-	// reconcile LoadBalancer
-	clusterScope.Info("Reconciling VPC load balancers")
-	if requeue, err := clusterScope.ReconcileLoadBalancers(); err != nil {
-		clusterScope.Error(err, "failed to reconcile VPC load balancers")
-		conditions.MarkFalse(powerVSCluster, infrav1beta2.LoadBalancerReadyCondition, infrav1beta2.LoadBalancerReconciliationFailedReason, capiv1beta1.ConditionSeverityError, err.Error())
-		return reconcile.Result{}, err
-	} else if requeue {
-		clusterScope.Info("VPC load balancer creation is pending, requeuing")
-		return reconcile.Result{RequeueAfter: 1 * time.Minute}, nil
-	}
+	conditions.MarkTrue(updatePowerVSCluster.cluster, infrav1beta2.TransitGatewayReadyCondition)
 
 	// reconcile COSInstance
 	if clusterScope.IBMPowerVSCluster.Spec.Ignition != nil {
 		clusterScope.Info("Reconciling COS service instance")
 		if err := clusterScope.ReconcileCOSInstance(); err != nil {
-			conditions.MarkFalse(powerVSCluster, infrav1beta2.COSInstanceReadyCondition, infrav1beta2.COSInstanceReconciliationFailedReason, capiv1beta1.ConditionSeverityError, err.Error())
+			conditions.MarkFalse(updatePowerVSCluster.cluster, infrav1beta2.COSInstanceReadyCondition, infrav1beta2.COSInstanceReconciliationFailedReason, capiv1beta1.ConditionSeverityError, err.Error())
 			return reconcile.Result{}, err
 		}
-		conditions.MarkTrue(powerVSCluster, infrav1beta2.COSInstanceReadyCondition)
+		conditions.MarkTrue(updatePowerVSCluster.cluster, infrav1beta2.COSInstanceReadyCondition)
+	}
+
+	var networkReady, loadBalancerReady bool
+	for _, cond := range clusterScope.IBMPowerVSCluster.Status.Conditions {
```

Comment:
By this time network should be ready right, When do you think it may not be ready also if its not ready we cannot even attach with TransitGateway right?

Reply:
should we consider moving this check before calling `ReconcileTransitGateway` to avoid failure?

Reply:
With transit gateway we are directly attaching the powervs with CRN right? It is not dependent on the network. And I m not requeuing when network and load balancer is not ready since it would block the TG creation even though it is not dependant for the TG. But before proceeding to retrieve the VPC LB details and set the overall Ready status as true, wanted to validate their status has reached Ready, that's why added the condition to validate both of these resource's Ready status.